### DISCUSSION TITLE
the esp32 does not map the buttons

### DISCUSSION THREAD

**Discussion by Ilikeconsoles** [darthcloud/BlueRetro#220] [Q&A]
hello good morning or good evening, I have a problem with the esp32 does not map the buttons It is configured for ps2 
 hola buenos dias o buenas noches, tengo un problema con el esp32 no mapea los botones está configurado para ps2

**Comment by darthcloud** [comment]
Posts pictures of your install.

**Comment by Ilikeconsoles** [comment]
![IMG-20220313-WA0003](https://user-images.githubusercontent.com/101438106/158087666-910e3a62-67c1-4a7b-9b97-227b7b4ab14d.jpg)

**Reply by darthcloud** [reply]
Hard to see but make sur you solder joint are good but checking continuity with multimeter between the bottom header pin and the respective PS plug pin.

**Reply by Ilikeconsoles** [reply]
Ya le cambie los cables por unos mejores y todavía sigue igual no mapea lo botones y también aveces un se queda presionado un botón y tengo que reiniciarlo a cada rato para que agarre que puedo hacer

**Reply by NickGravley** [reply]
It looks like you are using USB for powering the ESP32.  In this case you may still need to wire GND pin from the controller port to a GND pin for the ESP32, so the SPI signal have better reference.  The PS2 model I have outputs about 3.4V on controller ports, which is close enough 3.3v for direct connection to ESP32 VCC (3.3V) pin.  It shows steady comnunication on padtest.elf when wired this way (without usb power.)

**Reply by Ilikeconsoles** [reply]
![IMG_20220326_181959](https://user-images.githubusercontent.com/101438106/160746239-a95997d7-69c6-4211-a153-4357d36dfc38.jpg)

**Reply by Ilikeconsoles** [reply]
The fan is decorative to make it look good.

It is connected to 8 volts

**Comment by NickGravley** [comment]
Have you already verified (with a multimeter) the wire continuities, the absence of shorts,  the 3.3v at d5,d26,d27 regardless of fw,  and the various necessary grounded pins if using the universal fw?
